modules/nlu/src/views/full/LiteEditor.tsx

```diff
@@ -0,0 +1,85 @@
+import { Button, ControlGroup, FormGroup } from '@blueprintjs/core'
+import { NLU } from 'botpress/sdk'
+import React, { FC, useEffect, useState } from 'react'
+
+import { makeApi } from '../api'
+import { IntentEditor } from '../lite/intentEditor/IntentEditor'
+
+import style from './style.scss'
+import IntentDropdown from './IntentDropdown'
+import NameModal from './NameModal'
+
+interface IntentParams {
+  intentName: string
+}
+
+interface Props {
+  bp: any
+  contentLang: string
+  params: IntentParams
+  updateParams: (params: IntentParams) => void
+}
+
+export const LiteEditor: FC<Props> = props => {
+  const [intents, setIntents] = useState<NLU.IntentDefinition[]>([])
+  const [currentIntent, setCurrentIntent] = useState(props.params.intentName)
+  const [isModalOpen, setModalOpen] = useState(false)
+
+  const api = makeApi(props.bp)
+
+  useEffect(() => {
+    // tslint:disable-next-line: no-floating-promises
+    loadIntents()
+  }, [])
+
+  const loadIntents = async () => {
+    setIntents(await api.fetchIntents())
+  }
+
+  const createIntent = async (sanitizedName: string) => {
+    const intentDef = {
+      name: sanitizedName,
+      utterances: { [props.contentLang]: [name] }
+    }
+
+    props.updateParams({ intentName: sanitizedName })
+    await api.createIntent(intentDef)
+    await loadIntents()
+
+    setCurrentIntent(sanitizedName)
+  }
+
+  const onIntentChanged = intent => {
+    if (intent) {
+      setCurrentIntent(intent.name)
+      props.updateParams({ intentName: intent.name })
+    }
+  }
+
+  const toggleModal = () => setModalOpen(!isModalOpen)
+
+  return (
+    <div>
+      <NameModal isOpen={isModalOpen} toggle={toggleModal} intents={intents} onCreate={createIntent} />
+      {currentIntent && (
+        <IntentEditor
+          liteEditor={true}
+          intent={currentIntent}
+          api={api}
+          contentLang={props.contentLang}
+          showSlotPanel={false}
```

Comment:
I guess every `xyz={false}` is not needed, also `={true}` is redundant, as per `EFF` feedback earlier :)

Reply:
True, but it depends on the typing of the referenced component. In that case, it expects a boolean, and throws an error if it is left out. I changed it to optional